### ISSUE TITLE
InspectorServer: crashes when closing SystemMonitor: ASSERTION FAILED: may_defer || !called_from_inside_function

### ISSUE THREAD

**Issue by bcoles** [SerenityOS/serenity#13505]
1. Open `SystemMonitor`
2. Close the System Monitor window
3. :fire: 

<details>

```
18.587 InspectorServer(11:11): ASSERTION FAILED: may_defer || !called_from_inside_function
../.././AK/Function.h:194
18.595 [#0 InspectorServer(11:11)]: Terminating InspectorServer(11) due to signal 6
18.601 [#0 FinalizerTask(4:4)]: Backtrace:
0xdeadc0de  Kernel::Processor::switch_context(Kernel::Thread*&, Kernel::Thread*&) + 0x31e
0xdeadc0de  Kernel::Scheduler::context_switch(Kernel::Thread*) [clone .localalias] + 0x1f2
0xdeadc0de  Kernel::Scheduler::pick_next() [clone .localalias] + 0xfe
0xdeadc0de  Kernel::Processor::check_invoke_scheduler() [clone .localalias] + 0x175
0xdeadc0de  Kernel::Thread::die_if_needed() [clone .localalias] + 0x32d
0xdeadc0de  syscall_handler + 0x13ac
0xdeadc0de  syscall_asm_entry + 0x31
0x0b782055  libc.so: .text + 0x1b055
0x0b768c2a  libc.so: .text + 0x1c2a
0x0b76e84d  libc.so: .text + 0x784d
0xab5b653f  /bin/InspectorServer: .text + 0x853f
0xab5b75bf  /bin/InspectorServer: .text + 0x95bf
0x3619c1e9  libcore.so.serenity: .text + 0x451e9
0x3618d68a  libcore.so.serenity: .text + 0x3668a
0x3618f391  libcore.so.serenity: .text + 0x38391
0x36177695  libcore.so.serenity: .text + 0x20695
0x36177ffb  libcore.so.serenity: .text + 0x20ffb
0xab5b589f  /bin/InspectorServer: .text + 0x789f
0xab5ae618  /bin/InspectorServer: .text + 0x618
0xab5ae846  /bin/InspectorServer: .text + 0x846
18.636 [#0 FinalizerTask(4:4)]: Generating coredump for pid: 11
18.659 CrashDaemon(9:9): New coredump file: /tmp/coredump/InspectorServer_11_1649073699
18.667 SystemServer(6:6): Service InspectorServer has exited with exit code 6
18.812 CrashReporter(39:39): Started thread "", tid = 41
22.258 CrashReporter(39:41): Generating backtrace took 3426 ms
22.270 CrashReporter(39:41): --- Backtrace for thread #0 (TID 11) ---
22.296 CrashReporter(39:41): 0x984fe02e: [libsystem.so] syscall2 +0xe (syscall.cpp:25 => syscall.cpp:24)
22.324 CrashReporter(39:41): 0x0b782054: [libc.so] raise +0x24 (syscall.h:35 => signal.cpp:22 => signal.cpp:37)
22.335 CrashReporter(39:41): 0x0b768c29: [libc.so] abort +0x29 (stdlib.cpp:223)
22.358 CrashReporter(39:41): 0x0b76e84c: [libc.so] __assertion_failed +0x6c (assert.cpp:33)
22.380 CrashReporter(39:41): 0xab5b653e: [/bin/InspectorServer] AK::NonnullOwnPtr<InspectorServer::InspectableProcess>::clear() +0x12e (Function.h:194 => Function.h:61 => Stream.h:92 => Stream.h:453 => NonnullOwnPtr.h:147 => NonnullOwnPtr.h:54 => InspectableProcess.h:16)
22.397 CrashReporter(39:41): 0xab5b75be: [/bin/InspectorServer] AK::Function<void ()>::CallableWrapper<InspectorServer::InspectableProcess::InspectableProcess(int, AK::NonnullOwnPtr<Core::Stream::LocalSocket>)::{lambda()#1}>::call() +0x15e (NonnullOwnPtr.h:54 => HashMap.h:19 => HashTable.h:699 => HashTable.h:417 => HashMap.h:62 => InspectableProcess.cpp:30 => Function.h:151)
22.434 CrashReporter(39:41): 0x3619c1e8: [libcore.so.serenity] AK::Function<void ()>::CallableWrapper<Core::Stream::LocalSocket::setup_notifier()::{lambda()#1}>::call() +0x58 (Function.h:91)
22.450 CrashReporter(39:41): 0x3618d689: [libcore.so.serenity] Core::Notifier::event(Core::Event&) [clone .localalias] +0x99 (Function.h:91)
22.472 CrashReporter(39:41): 0x3618f390: [libcore.so.serenity] Core::Object::dispatch_event(Core::Event&, Core::Object*) +0xb0 (Object.cpp:220)
22.499 CrashReporter(39:41): 0x36177694: [libcore.so.serenity] Core::EventLoop::pump(Core::EventLoop::WaitMode) [clone .localalias] +0x154 (EventLoop.cpp:481)
22.509 CrashReporter(39:41): 0x36177ffa: [libcore.so.serenity] Core::EventLoop::exec() +0x11a (EventLoop.cpp:438)
22.522 CrashReporter(39:41): 0xab5b589e: [/bin/InspectorServer] serenity_main(Main::Arguments) +0x1ee (main.cpp:32)
22.526 CrashReporter(39:41): 0xab5ae617: [/bin/InspectorServer] main +0x77 (Main.cpp:39)
22.541 CrashReporter(39:41): 0xab5ae845: [/bin/InspectorServer] _entry +0x65 (crt0.cpp:49)
```

</details>

**Comment by awesomekling** [comment]
cc @kleinesfilmroellchen

**Comment by kleinesfilmroellchen** [comment]
This appears to be a general IS problem; nobody makes itself available to IS except SystemMonitor.

**Comment by sin-ack** [comment]
Fixed by #13535.

**Comment by bcoles** [comment]
Confirmed #13535 resolved this issue.

Although on a possibly unrelated note, `SystemMonitor` now stops responding when more than one instance is run concurrently.

![SystemMonitor](https://user-images.githubusercontent.com/434827/161990379-bd3f213b-fc79-4345-83c8-e732c9bc06d1.png)

Navigating to the `Performance` tab seems to be required to reproduce this issue. The `SystemMonitor` does not unfreeze once the second instance is closed.
